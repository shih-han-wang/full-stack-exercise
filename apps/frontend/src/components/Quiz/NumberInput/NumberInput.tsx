import {
  NumberDecrementStepper,
  NumberIncrementStepper,
  NumberInput,
  NumberInputField,
  NumberInputStepper
} from '@chakra-ui/react'

type Props = {
  value: number
  disabled?: boolean
  onChange: () => void
}

const Number = ({ value, disabled = false, onChange }: Props) => (
  <NumberInput value={value} onChange={onChange} isDisabled={disabled}>
    <NumberInputField />
    <NumberInputStepper>
      <NumberIncrementStepper />
      <NumberDecrementStepper />
    </NumberInputStepper>
  </NumberInput>
)

export default Number
