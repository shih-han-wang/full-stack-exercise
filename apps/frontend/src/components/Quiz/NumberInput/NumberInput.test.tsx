import { render } from '@testing-library/react'
import NumberInput from './NumberInput'

const onChangeMock = jest.fn()

describe('Components > Quiz > NumberInput', () => {
  it('should render correctly', async () => {
    const view = render(<NumberInput value={5} onChange={onChangeMock} />)

    expect(view.asFragment()).toMatchSnapshot()
  })

  it('should render disabled state correctly', async () => {
    const view = render(
      <NumberInput value={5} onChange={onChangeMock} disabled />
    )

    expect(view.asFragment()).toMatchSnapshot()
  })
})
