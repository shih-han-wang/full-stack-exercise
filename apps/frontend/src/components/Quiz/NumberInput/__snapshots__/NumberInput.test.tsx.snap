// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Quiz > NumberInput should render correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-numberinput css-0"
    value="5"
  >
    <input
      aria-valuemax="9007199254740991"
      aria-valuemin="-9007199254740991"
      aria-valuenow="5"
      aria-valuetext="5"
      autocomplete="off"
      autocorrect="off"
      class="chakra-numberinput__field css-0"
      inputmode="decimal"
      pattern="[0-9]*(.[0-9]+)?"
      role="spinbutton"
      type="text"
      value="5"
    />
    <div
      aria-hidden="true"
      class="css-0"
    >
      <div
        class="css-0"
        role="button"
        tabindex="-1"
      >
        <svg
          class="chakra-icon css-0"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="css-0"
        role="button"
        tabindex="-1"
      >
        <svg
          class="chakra-icon css-0"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Components > Quiz > NumberInput should render disabled state correctly 1`] = `
<DocumentFragment>
  <div
    class="chakra-numberinput css-0"
    value="5"
  >
    <input
      aria-valuemax="9007199254740991"
      aria-valuemin="-9007199254740991"
      aria-valuenow="5"
      aria-valuetext="5"
      autocomplete="off"
      autocorrect="off"
      class="chakra-numberinput__field css-0"
      disabled=""
      inputmode="decimal"
      pattern="[0-9]*(.[0-9]+)?"
      role="spinbutton"
      type="text"
      value="5"
    />
    <div
      aria-hidden="true"
      class="css-0"
    >
      <div
        aria-disabled="true"
        class="css-0"
        disabled=""
        role="button"
        tabindex="-1"
      >
        <svg
          class="chakra-icon css-0"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12.8,5.4c-0.377-0.504-1.223-0.504-1.6,0l-9,12c-0.228,0.303-0.264,0.708-0.095,1.047 C2.275,18.786,2.621,19,3,19h18c0.379,0,0.725-0.214,0.895-0.553c0.169-0.339,0.133-0.744-0.095-1.047L12.8,5.4z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        aria-disabled="true"
        class="css-0"
        disabled=""
        role="button"
        tabindex="-1"
      >
        <svg
          class="chakra-icon css-0"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M21,5H3C2.621,5,2.275,5.214,2.105,5.553C1.937,5.892,1.973,6.297,2.2,6.6l9,12 c0.188,0.252,0.485,0.4,0.8,0.4s0.611-0.148,0.8-0.4l9-12c0.228-0.303,0.264-0.708,0.095-1.047C21.725,5.214,21.379,5,21,5z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
